src/panels/config/scene/ha-scene-editor.ts

```diff
@@ -634,7 +635,15 @@ export class HaSceneEditor extends SubscribeMixin(
       if (!entityReg.device_id) {
         continue;
       }
-      if (!this._devices.includes(entityReg.device_id)) {
+      const entity = config.entities[entityReg.entity_id];
+      const entityMetaData = config.metadata?.[entityReg.entity_id];
+      if (
+        !this._devices.includes(entityReg.device_id) &&
+        !(typeof entity === "string") &&
+        entityMetaData
+          ? !entityMetaData.entity_only
+          : false
```

Comment:
It took me a while staring at this code to figure out what it does. This seems to simplify it a bit. 
```suggestion
        entityMetaData && 
        !entityMetaData.entity_only
```

Reply:
To not change the behavior in a surprising way, the logic should be reversed so we assume entities in scenes without metadata are added as devices.